nodecontrol/algodControl.go

```diff
@@ -283,8 +288,18 @@ func (nc NodeController) GetAlgodPath() string {
 
 // Clone creates a new DataDir based on the controller's DataDir; if copyLedger is true, we'll clone the ledger.sqlite file
 func (nc NodeController) Clone(targetDir string, copyLedger bool) (err error) {
-	os.RemoveAll(targetDir)
-	err = os.Mkdir(targetDir, 0700)
+	err = os.RemoveAll(targetDir)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("unable to delete directory '%s' : %v", targetDir, err)
+	}
+
+	sourceFolderStat, statError := os.Stat(nc.algodDataDir)
```

Comment:
Why not use the err here instead of introducing  a new variable statError?

Reply:
I find that the go language syntax of 
```
a, b := func()
```
is misleading. It's not clear if b was defined before, or is defined within this line.
I'll find another way to do that.

Reply:
Oops. I missed that. It is fine the way it is.